libc/stdlib_strto.c

```diff
@@ -0,0 +1,671 @@
+	}
+}
+
+
+TEST(stdlib_strto, strtoll_basic_hex)
+{
+	const char *str[] = { "0x0", "0x1", "-0x499602D2", "0x499602D2", "0x7FFFFFFF", "-0x80000000" };
+	const long long expected[] = { 0, 1, -1234567890, 1234567890, INT_MAX, INT_MIN };
+
+	for (int i = 0; i < (sizeof(str) / sizeof(str[0])); i++) {
+		errno = 0;
+		TEST_ASSERT_EQUAL_INT(expected[i], strtoll(str[i], NULL, 16));
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+}
+
+
+TEST(stdlib_strto, strtoul_basic)
+{
+	const char *str[] = { "0", "1", "-1234567890", "1234567890", "2147483647", "-2147483648" };
+	const unsigned long expected[] = { 0, 1, -1234567890, 1234567890, INT_MAX, INT_MIN };
+
+	for (int i = 0; i < (sizeof(str) / sizeof(str[0])); i++) {
+		errno = 0;
+		TEST_ASSERT_EQUAL_INT(expected[i], strtoul(str[i], NULL, 10));
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+}
+
+
+TEST(stdlib_strto, strtoul_basic_binary)
+{
+	const char *str[] = { "0", "1", "10011010010", "-10011010010", "1111111111111111111111111111111", "-10000000000000000000000000000000" };
+	const unsigned long expected[] = { 0, 1, 1234, -1234, INT_MAX, INT_MIN };
+
+	for (int i = 0; i < (sizeof(str) / sizeof(str[0])); i++) {
+		errno = 0;
+		TEST_ASSERT_EQUAL_INT(expected[i], strtoul(str[i], NULL, 2));
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+}
+
+
+TEST(stdlib_strto, strtoul_basic_hex)
+{
+	const char *str[] = { "0x0", "0x1", "-0x499602D2", "0x499602D2", "0x7FFFFFFF" };
+	const unsigned long expected[] = { 0, 1, -1234567890, 1234567890, INT_MAX };
+
+	for (int i = 0; i < (sizeof(str) / sizeof(str[0])); i++) {
+		errno = 0;
+		TEST_ASSERT_EQUAL_INT(expected[i], strtoul(str[i], NULL, 16));
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+}
+
+
+TEST(stdlib_strto, strtoull_basic)
+{
+	const char *str[] = { "0", "1", "-1234567890", "1234567890", "2147483647", "-2147483648" };
+	const unsigned long long expected[] = { 0, 1, -1234567890, 1234567890, INT_MAX, INT_MIN };
+
+	for (int i = 0; i < (sizeof(str) / sizeof(str[0])); i++) {
+		errno = 0;
+		TEST_ASSERT_EQUAL_INT(expected[i], strtoull(str[i], NULL, 10));
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+}
+
+
+TEST(stdlib_strto, strtoull_basic_binary)
+{
+	const char *str[] = { "0", "1", "10011010010", "-10011010010", "1111111111111111111111111111111", "-10000000000000000000000000000000" };
+	const unsigned long long expected[] = { 0, 1, 1234, -1234, INT_MAX, INT_MIN };
+
+	for (int i = 0; i < (sizeof(str) / sizeof(str[0])); i++) {
+		errno = 0;
+		TEST_ASSERT_EQUAL_INT(expected[i], strtoull(str[i], NULL, 2));
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+}
+
+
+TEST(stdlib_strto, strtoull_basic_hex)
+{
+	const char *str[] = { "0x0", "0x1", "-0x499602D2", "0x499602D2", "0x7FFFFFFF" };
+	const unsigned long long expected[] = { 0, 1, -1234567890, 1234567890, INT_MAX };
+
+	for (int i = 0; i < (sizeof(str) / sizeof(str[0])); i++) {
+		errno = 0;
+		TEST_ASSERT_EQUAL_INT(expected[i], strtoull(str[i], NULL, 16));
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+}
+
+
+TEST(stdlib_strto, strto_inf_nan)
```

Comment:
I know that you wanted to be consistent with other names, but IMHO this prefix is not needed and that suggests test case applies to all `strto` functions

```suggestion
TEST(stdlib_strto, inf_nan)
```